tests/e2e/test_subshell.py

```diff
@@ -15,3 +15,85 @@ def test_subshell_with_pipeline(shell_session):
 
     result = get_command_output(shell_session.before)
     assert "Hello$" == result
+
+
```

Comment:
The tests below should be split into a separate file named like `test_shlvl.py`.